packages/ui/src/working-groups/components/Roles/RolesList.tsx

```diff
@@ -80,13 +97,14 @@ const RolesListItem = ({ worker }: { worker: Worker }) => {
         </OpenItemSummaryColumn>
         <OpenItemSummaryColumn>
           <TextInlineBig>
-            <TokenValue value={worker.rewardPerBlock} />
+            <TokenValue value={earnings} />
           </TextInlineBig>
           <ToggleableSubscriptionWide>Earned total</ToggleableSubscriptionWide>
         </OpenItemSummaryColumn>
         <OpenItemSummaryColumn>
           <TextInlineBig>
-            <TokenValue value={earnings} />
+            <NumericValue>{payout?.gte(BN_ZERO) ? payout?.toString() + ' blocks' : '–'}</NumericValue>
+            {/* {payout && <TextMedium lighter>({payout && blocksToTime(payout)})</TextMedium>} */}
```

Comment:
[getNextPayout](https://github.com/Joystream/pioneer/pull/738/commits/1dd175f01cb2749f5c7e1bff01a60c5628b6c45f) returns BN ("Big Number"). WDYT about renaming `payout` to `blocks` to make the code more clear?

We expect "payout" to be defined. Why the switch to `NumericValue` and then pass `-`? No need to re-implement [blocksToTime](https://github.com/Joystream/pioneer/blob/dev/packages/ui/src/common/model/blocksToTime.ts):
```suggestion
            <TextMedium lighter>{blocksToTime(payout)}</TextMedium>
```

Reply:
I guess it can be `undefined` when `blockNumber` is `undefined` (before the api connection/ before the block height is returned).

But I took a look at `getNextPayout` this function is weird... I don't see why it should return a `BN` to begin with (it's an amount of blocks it won't go over any js limits). It returns `BN(-1)` when no worker are passed :confused: and the reduce mutates it's accumulator. I think: `Math.max(0, Math.min(...nextPayoutPerGroup))` should work (given `nextPayoutPerGroup` becomes `number[]`)